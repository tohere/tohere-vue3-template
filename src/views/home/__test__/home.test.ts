import { mount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { describe, it, expect } from "vitest";
import Home from "../index.vue";
import ElementPlus from "element-plus";
import "element-plus/dist/index.css";

describe("Home Test", () => {
  let wrapper: VueWrapper;
  const props = {
    msg: "hello world",
  };
  beforeEach(() => {
    setActivePinia(createPinia());

    wrapper = mount(Home, {
      props,
      global: {
        plugins: [ElementPlus],
      },
    });
  });
  it("should render", () => {
    expect(wrapper.text()).toContain(props.msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("button click", async () => {
    const count = wrapper.find(".count");
    const btn = wrapper.find("button");
    expect(count.text()).toBe("0");
    await btn.trigger("click");
    expect(count.text()).toBe("1");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
