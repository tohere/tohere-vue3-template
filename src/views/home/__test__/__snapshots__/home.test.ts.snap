// Vitest Snapshot v1

exports[`Home Test > button click 1`] = `
"<div>Home</div>
<h1>----2</h1>
<h2 class=\\"count\\">1</h2>
<h2>msg: hello world</h2>
<button>增加</button>"
`;

exports[`Home Test > should render 1`] = `
"<div>Home</div>
<h1>----0</h1>
<h2 class=\\"count\\">0</h2>
<h2>msg: hello world</h2>
<button>增加</button>"
`;
